gui/system/forms.py

```diff
@@ -1710,19 +1711,72 @@ def __init__(self, *args, **kwargs):
                 pool_choices.append((v.vol_name, v.vol_name))
 
         self.fields['sys_pool'].choices = pool_choices
-        self.instance._original_sys_pool = self.instance.sys_pool
-        self.instance._original_sys_syslog_usedataset = self.instance.sys_syslog_usedataset
-        self.instance._original_sys_rrd_usedataset = self.instance.sys_rrd_usedataset
         self.fields['sys_pool'].widget.attrs['onChange'] = (
             "systemDatasetMigration();"
         )
 
     def middleware_clean(self, update):
         update['syslog'] = update.pop('syslog_usedataset')
-        update['rrd'] = update.pop('rrd_usedataset')
         return update
 
 
+class ReportingForm(MiddlewareModelForm, ModelForm):
+
+    middleware_attr_prefix = ""
+    middleware_attr_schema = "reporting"
+    middleware_plugin = "reporting"
+    is_singletone = True
+
+    rrd_size_alert_threshold = SizeField(
+        required=False,
+        label=_("Reporting database size alert threshold"),
+        help_text=_(
+            "Store reporting database in system dataset instead of RAMDisk. Checking this will decrease RAM usage and "
+            "increase IO on system dataset pool."
+        ),
+    )
+    rrd_ramdisk_size = SizeField(
+        label=_("Reporting database RAMDisk size"),
+    )
+    graph_timespans = forms.CharField(
+        label=_("Graph time spans"),
+        help_text=_("Time periods for which aggregated historical data will be stored. See collect RRARows option "
+                    "documentation for more details"),
```

Comment:
In new UI I was thinking of a user-friendly widget where user can say "I want graphs for 1 day, 1 week, 2 months, 5 years". Also I don't think they'll get bothered with designing it considering it's an advanced option for 0.01% of users.

Probably we'll need to keep to plain text widget with text like "1h 1d 1w 1m 1y" and so in new UI. Do you think it's user-friendly enough?

Reply:
I am starting to wonder if there is any real value in exposing it then?

We use that data to accurately show them the data in Reporting UI, does changing that would change the behavior of that and/or how the API to retrieve that data is used?

Reply:
Then perhaps we should focus on that attribute alone, size? I am trying to think from a user perspective and I am not sure mimic the collectd knob would be the best approach, perhaps we could just have one attribute saying for how long the reporting should be kept (perhaps in months, defaulting to 12) then on middleware we convert that to whatever makes most sense in collectd?

I say this for two reasons, we should try to abstract implementation details as much as possible from the end user and because collectd may not be always the our implementation (its possible we try and move to netdata someday)

Reply:
This should have been discussed in the design doc, sorry about that, but this only came to me now ; )